lib/player.js

```diff
@@ -4835,23 +4836,29 @@ shaka.Player = class extends shaka.util.FakeEventTarget {
     const data = shaka.util.BufferUtils.toUint8(buffer);
     const cues = TextParser.parseMedia(data, time, uri);
 
+    const baseUri = new goog.Uri(uri);
     const references = [];
     for (const cue of cues) {
-      const imageUri = shaka.util.ManifestParserUtils.resolveUris(
-          [uri], [cue.payload])[0];
+      let uris = null;
+      const getUris = () => {
+        if (uris == null) {
```

Comment:
I understand why not resolving all of the uris at once in a huge thumbnails track might improve performance, but is there a specific reason we want to no longer use `shaka.util.ManifestParserUtils.resolveUris` for this? It seems like it'd be a smaller change for getUris to just be something like:

```js
const getUris = () => {
  if (uris == null) {
    uris = shaka.util.ManifestParserUtils.resolveUris([uri], [cue.playload]);
  }
  return uris || [];
};
```

Reply:
Ok, I’ll change it